File: src/main/java/br/com/controllers/Usuario/UsuarioController.java
package br.com.controllers.usuario;

import java.util.Map;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import br.com.model.entities.classes.usuario.Usuario;
import br.com.services.classes.usuario.UsuarioService;

@RestController
@CrossOrigin(origins = "*")
public class UsuarioController extends UsuarioService {

    private final String URL = "/usuario";

    @GetMapping(URL + "/all")
    protected ResponseEntity<Iterable<Usuario>> procurarTodos() {
        return super.procurarTodos();
    }

    @PostMapping(URL)
    protected ResponseEntity<Usuario> salvar(@RequestBody Usuario usuario) throws Exception {
        return super.salvar(usuario);
    }

    @DeleteMapping(URL)
    protected ResponseEntity<String> remover(@RequestBody Usuario usuario) throws Exception {
        return super.remover(usuario);
    }

    @PatchMapping(URL)
    protected ResponseEntity<Usuario> atualizar(@RequestBody Usuario usuario) throws Exception {
        return super.atualizar(usuario);
    }

    @GetMapping(URL + "/{id}")
    protected ResponseEntity<Usuario> procurarPorID(@PathVariable Integer id) {
        return super.procurarPorID(id);
    }

    @GetMapping(URL + "/buscar")
    private ResponseEntity<Usuario> buscar(@RequestBody Map<String, String> cnpjCpf) {
        return super.buscar(cnpjCpf.get("cnpjCpf"));
    }

    @PostMapping(URL + "/AutenticarUsuario/")
    protected Usuario autenticaUsuario(@RequestBody Usuario usuario) {
        return super.autenticaUsuario(usuario);
    }

}

File: src/main/java/br/com/controllers/usuario/UsuarioController.kt
package br.com.controllers.usuario

import br.com.repository.usuario.UsuarioRepository
import br.com.services.classes.usuario.UsuarioService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import br.com.model.usuario.Usuario

@RestController
@CrossOrigin(origins = ["*"])
class UsuarioController(@Autowired private val usuarioRepository: UsuarioRepository) : UsuarioService
    (usuarioRepository) {

    companion object {
        const val URL = "/usuario"
    }

    @GetMapping("$URL/all")
    override fun procurarTodos(): ResponseEntity<MutableIterable<Usuario>> {
        return super.procurarTodos()
    }

    @PostMapping(URL)
    @Throws(Exception::class)
    override fun salvar(@RequestBody usuario: Usuario): ResponseEntity<Usuario> {
        return super.salvar(usuario)
    }

    @DeleteMapping(URL)
    @Throws(Exception::class)
    override fun remover(@RequestBody usuario: Usuario): ResponseEntity<String> {
        return super.remover(usuario)
    }

    @PatchMapping(URL)
    @Throws(Exception::class)
    override fun atualizar(@RequestBody usuario: Usuario): ResponseEntity<Usuario> {
        return super.atualizar(usuario)
    }

    @GetMapping("$URL/{id}")
    override fun procurarPorID(@PathVariable id: Int): ResponseEntity<Usuario> {
        return super.procurarPorID(id)
    }

    @GetMapping("$URL/buscar")
    override fun buscar(@RequestBody cnpjCpf: String): ResponseEntity<Usuario> {
        return super.buscar(cnpjCpf)
    }

    @PostMapping("$URL/AutenticarUsuario/")
    override fun autenticaUsuario(@RequestBody usuario: Usuario): Usuario? {
        return super.autenticaUsuario(usuario)
    }
}
